pdb2sql/superpose.py

```diff
@@ -0,0 +1,278 @@
+
+import os
+import numpy as np
+from .pdb2sqlcore import pdb2sql
+from .transform import rotate
+
+
+def superpose(mobile, target, method='svd', only_backbone=True, export=True, **kwargs):
+    """superpose two complexes
+
+    Arguments:
+        mobile {str or pdb2sql} -- name or sqldb of the mobile pdb
+        target {str or pdb2sql} -- name or sqldb of the target pdb
+
+    Keyword Arguments:
+        method {str} -- method used to superpose the complex (default: {'svd'})
+        only_backbone {bool} -- use only backbone atos to align (default: True)
+        **kwargs -- keyword arguments used to select a portion of the pdb
+
+    Example:
+        >> pdb1 = '1AK4_5w.pdb'
+        >> pdb2 = '1AK4_10w.pdb'
+        >> superpose(pdb1, pdb2, chainID='A')
+
+    """
+
+    backbone_atoms = ['CA', 'C', 'N', 'O']
+
+    if not isinstance(mobile, pdb2sql):
+        sql_mobile = pdb2sql(mobile)
+    else:
+        sql_mobile = mobile
+
+    if not isinstance(target, pdb2sql):
+        sql_target = pdb2sql(target)
+    else:
+        sql_target = target
+
+    if only_backbone:
+        if 'name' not in kwargs:
+            kwargs['name'] = backbone_atoms
+        else:
+            raise ValueError('Atom type specified but only_backbone == True')
+
+    # selections of some atoms
+    selection_mobile = np.array(sql_mobile.get("x,y,z", **kwargs))
+    selection_target = np.array(sql_target.get("x,y,z", **kwargs))
+
+    # the molbile original coordinates
+    xyz_mobile = np.array(sql_mobile.get("x,y,z"))
+
+    # transform the xyz mobile
+    xyz_mobile = transform_xyz_from_selection_superposition(xyz_mobile,
+                                                            selection_mobile, selection_target, method)
+
+    # update the sql
+    sql_mobile.update('x,y,z', xyz_mobile)
+
+    # export a pdb file
+    if export:
+        target_name = os.path.basename(sql_target.pdbfile).rstrip('.pdb')
+        mobile_name = os.path.basename(sql_mobile.pdbfile).rstrip('.pdb')
+        fname = mobile_name + '_superposed_on_' + \
+            target_name + '.pdb'
+        sql_mobile.exportpdb(fname)
+
+    return sql_mobile
+
+
+def transform_xyz_from_selection_superposition(xyz_mobile,
```

Comment:
what do you think changing the the name to `superpose_xyz`?

Reply:
it's not what it does exactly ... i superpose 2 fragments and transform the entire structure .... 
superpose_fragment ?

Reply:
changed to `superpose_selection`

Reply:
Great, thanks.